internal/key/key.go

```diff
@@ -29,191 +30,270 @@ const (
 	Size = 256 / 8
 )
 
-// Key is a secret key for symmetric cryptography.
-type Key struct {
-	bytes []byte
-}
+// Algorithm is a cryptographic algorithm that requires
+// a cryptographic key.
+type Algorithm string
```

Comment:
I don't know how size sensitive you are, but obviously a uint8 would take less space - and you can autogen the stringer. Validation is easier.

Reply:
Yes, that is something I can explore. Since we use a binary format we can actually also avoid the "expensive" string.

Reply:
However, that change needs some additional code changes.

Reply:
If you add a MarshalText or MarshalJSON you could maybe make it backwards compatible that way?